Compile error installing on AIX 7.2

Trying to install Pyinstaller on AIX 7.2. When I run python3 -m pip install Pyinstaller it gives the following error when it tries to compile the bootloader:

>   [14/21] Compiling src/pyi_path.c
  ../../src/pyi_pythonlib.c: In function 'pyi_pylib_load':
  ../../src/pyi_pythonlib.c:71:13: error: 'dllname' undeclared (first use in this function); did you mean 'dll_name'?
     71 |             dllname,
        |             ^~~~~~~
        |             dll_name
  ../../src/pyi_pythonlib.c:71:13: note: each undeclared identifier is reported only once for each function it appears in
  ../../src/pyi_pythonlib.c:72:13: error: 'DLLNAME_LEN' undeclared (first use in this function); did you mean 'MAX_DLL_NAME_LEN'?
     72 |             DLLNAME_LEN,
        |             ^~~~~~~~~~~
        |             MAX_DLL_NAME_LEN
  

If I make the changes suggested by the compiler and manually compile it then the errors go away. I don't know if this is the correct solution.

--- a/bootloader/src/pyi_pythonlib.c
+++ b/bootloader/src/pyi_pythonlib.c
@@ -68,8 +68,8 @@ pyi_pylib_load(struct PYI_CONTEXT *pyi_ctx)
         pyver_minor = archive->python_version % 100;
 
         dll_name_len = snprintf(
-            dllname,
-            DLLNAME_LEN,
+            dll_name,
+            MAX_DLL_NAME_LEN,
             "libpython%d.%d.a(libpython%d.%d.so)",
             pyver_major,
             pyver_minor,

--- a/news/8819.bootloader.rst
+++ b/news/8819.bootloader.rst
@@ -1,0 +1,2 @@
+(AIX) Fix errors when compiling bootloader under AIX (regression
+introduced in PyInstaller v6.8).

